### ISSUE TITLE
Bradley-Terry dataset token truncation

### ISSUE THREAD

**Issue by LazaroHurtado** [natolambert/rlhf-book#503]
Currently the BT reward model training script truncates to `max_length`. Depending on the prompt input + the chosen and rejected prefix it is possible that post-truncation the token ids are the same. In this scenario, the gradient of the loss will be zero so no real learning is happening. So, when we are calling `build_preference_dataset` should we drop rows where `chosen_tokens == rejected_tokens` or should we left truncated? If we left truncated then its possible the user prompt gets dropped so the hidden state of EOS token wont have that info when being passed to the reward model head.

Example:
```python
from datasets import load_dataset

from reward_models.base import load_tokenizer
from reward_models.config import Config
from reward_models.train_preference_rm import tokenize_messages

config = Config()
tokenizer = load_tokenizer(config.model_id)
row = load_dataset(
    config.dataset_name,
    split=config.dataset_split,
)[54257]

chosen = tokenize_messages(tokenizer, row["chosen"], max_length=512)
rejected = tokenize_messages(tokenizer, row["rejected"], max_length=512)

print(chosen["input_ids"] == rejected["input_ids"])  # True
```

**Comment by natolambert** [comment]
Yeah that sgtm, is the right minimal solution for this repo, I'm making the change.
